src/common/contentPreference.ts

```diff
@@ -0,0 +1,135 @@
+import { AuthContext } from '../Context';
+import {
+  ContentPreferenceStatus,
+  ContentPreferenceType,
+} from '../entity/contentPreference/types';
+import { ContentPreferenceUser } from '../entity/contentPreference/ContentPreferenceUser';
+import { NotificationPreferenceUser } from '../entity/notifications/NotificationPreferenceUser';
+import {
+  NotificationPreferenceStatus,
+  NotificationType,
+} from '../notifications/common';
+
+type FollowEntity = ({
+  ctx,
+  id,
+  status,
+}: {
+  ctx: AuthContext;
+  id: string;
+  status: ContentPreferenceStatus.Follow | ContentPreferenceStatus.Subscribed;
+}) => Promise<void>;
+
+type UnFollowEntity = ({
+  ctx,
+  id,
+}: {
+  ctx: AuthContext;
+  id: string;
+}) => Promise<void>;
+
+const entityToNotificationTypeMap: Record<
+  ContentPreferenceType,
+  NotificationType[]
+> = {
+  [ContentPreferenceType.User]: [NotificationType.UserPostAdded],
+};
+
+const followUser: FollowEntity = async ({ ctx, id, status }) => {
+  await ctx.con.transaction(async (entityManager) => {
+    const repository = entityManager.getRepository(ContentPreferenceUser);
+
+    const contentPreference = repository.create({
+      userId: ctx.userId,
+      referenceId: id,
+      referenceUserId: id,
+      status,
+    });
+
+    await repository.save(contentPreference);
+
+    const notificationRepository = entityManager.getRepository(
+      NotificationPreferenceUser,
+    );
+
+    if (status === ContentPreferenceStatus.Subscribed) {
+      const notificationPreferences = entityToNotificationTypeMap.user.map(
+        (notificationType) => {
+          return notificationRepository.create({
+            userId: ctx.userId,
+            referenceId: id,
+            referenceUserId: id,
+            notificationType,
+            status: NotificationPreferenceStatus.Subscribed,
+          });
+        },
+      );
+
+      await notificationRepository.save(notificationPreferences);
+    } else {
+      await notificationRepository.delete({
```

Comment:
Nothing major, but wondering if we should abstract this into a in file function, just for the sake if next month you need to add another field here.

Reply:
done